### ISSUE TITLE
Issue with reshape after reduce mean operation.

### ISSUE THREAD

**Issue by sri-oc** [onnx/tensorflow-onnx#514]
Hello,

I'm using resnet_v1_fp32_savedmodel_NHWC with the following model property, 

`inputs['input'] tensor_info:
        dtype: DT_FLOAT
        shape: (64, 224, 224, 3)
        name: input_tensor:0
  The given SavedModel SignatureDef contains the following output(s):
    outputs['classes'] tensor_info:
        dtype: DT_INT64
        shape: (64)
        name: ArgMax:0
    outputs['probabilities'] tensor_info:
        dtype: DT_FLOAT
        shape: (64, 1001)
        name: softmax_tensor:0
  Method name is: tensorflow/serving/predict
`

**Issue:**
o _output_shapes attribute of a mean operator in tensorflow frozen graph protobuf is of dimensions [64,1,1,2048]. 

o The output dimensions of Reduced mean in onnx modelproto after the conversion is [1, 2048, 1, 1].

o The dimension of shape input for reshape operator is  [64,1,1,2048]. This doesn't match with the total size of the reduced mean output dimension. Tensorflow-Onnx is not complaining about it. 

**Onnx model after conversion:**
![Screen Shot 2019-05-13 at 5 41 01 PM](https://user-images.githubusercontent.com/36420677/57664249-44c38600-75ac-11e9-9ad3-9600054baeb2.png)

**Tensoflow Frozen Graph**
![Screen Shot 2019-05-13 at 6 25 38 PM](https://user-images.githubusercontent.com/36420677/57664334-b7ccfc80-75ac-11e9-81f7-fca3e70b86c4.png)

Any suggestions are helpful. 

**Edit:05/15**
o When I'm passing converted ONNX model to Glow, it's complaining about the size mismatch.

**Comment by nbcsm** [comment]
could you share your model please? thanks.

**Comment by sri-oc** [comment]
@nbcsm  I'm afraid I can't upload the models as each model size is around 100MB. 

I took my model from official pre-trained models provided by tensorflow, [ResNet-50 v1 (fp32) NHWC](https://github.com/tensorflow/models/tree/master/official/resnet#pre-trained-model) and converted to Onnx  after freezing the model using the following command,

`
python3 -m tf2onnx.convert --input $PWD/frozen_graph.pb --inputs "input_tensor:0" --outputs "softmax_tensor:0" --output $PWD/model.onnx --verbose --opset 7 --inputs-as-nchw input_tensor:0
`

**Comment by nbcsm** [comment]
thanks, we will look into it.

**Comment by guschmue** [comment]
this is the same model we use for mlperf. The difference is that for mlperf we re-export the graph with batch-size=-1 and there is no problem with that model. I think that was the same case for the unmodified model. I ran 
```python tests/run_pretrained_models.py --tests resnet50_v1```
which is the same model and it passes.
This model wants a fixed batch-size of 64 and if you pass batch-size=1 it will fail, that is why we re-exported it for mlperf.

**Comment by sri-oc** [comment]
I can see that this model already has a batch size of 64. And, tf2onnx did not complain about the batch size or dimensions of reshape and reduce mean.  I think it shall complain unless I'm missing something over here. I'm hitting this issue once I pass the converted model to Glow.

**Comment by guschmue** [comment]
perfectly legal to use a batch-size of 64 in the model, just not common. As long you feed batches of 64 it is fine but a batch-size != 64 will fail. If you'd run it in tensorflow it will have the same issue. I'm not sure why this was chosen for this model.

The resnet50 onnx model for mlperf is the same model and we exported it with a batch-size of -1 for exactly the that reason. The mlperf model is here: https://github.com/mlperf/inference/tree/master/cloud/image_classification and the tools directory contains the scripts we used to generate it.
